core/app/models/spree/order.rb

```diff
@@ -117,19 +117,14 @@ def states
     before_create :link_by_email
     before_update :homogenize_line_item_currencies, if: :currency_changed?
 
-    validates :number,               presence:     true,
-                                     length:       { maximum: 32, allow_blank: true },
-                                     uniqueness:   { allow_blank: true }
-    validates :email,                presence:     true,
-                                     length:       { maximum: 254, allow_blank: true },
-                                     email:        { allow_blank: true },
-                                     if:           :require_email
-    validates :state,                presence:     true,
-                                     inclusion:    { in: state_machine.states.map { |state| state.name.to_s }, allow_blank: true }
+    with_options presence: true do
+      validates :number, length: { maximum: 32, allow_blank: true }, uniqueness: { allow_blank: true }
+      validates :email, length: { maximum: 254, allow_blank: true }, email: { allow_blank: true }, if: :require_email
+      validates :state, inclusion: { in: state_machine.states.map { |state| state.name.to_s }, allow_blank: true }
+      validates :item_count, numericality: { greater_than_or_equal_to: 0, less_than: 2**31, only_integer: true, allow_blank: true }
+    end
     validates :payment_state,        inclusion:    { in: %w[balance_due paid credit_owed failed void], allow_blank: true }
```

Comment:
Create array directly like ['balance_due', 'paid', 'etc' ] rather than %w[balance_due paid etc]
[] is faster that %w[]
 Benchmark.bm do |x|
1.9.3-p551 :009 >     x.report {  %w[balance_due paid credit_owed failed void] * 10000 }
1.9.3-p551 :010?>   x. report { ['balance_due' 'paid' 'credit_owed' 'failed' 'void'] * 10000}
1.9.3-p551 :011?>   end
       user     system      total        real
   0.000000   0.000000   0.000000 (  0.000105)
   0.000000   0.000000   0.000000 (  0.000024)
 => [  0.000000   0.000000   0.000000 (  0.000105)
,   0.000000   0.000000   0.000000 (  0.000024)
]